src/utils/UrlUtils.cpp

```diff
@@ -223,12 +223,31 @@ void UTILS::URL::AppendParameters(std::string& url, std::string_view params)
   if (params.empty())
     return;
 
-  if (url.find_first_of('?') == std::string::npos)
-    url += "?";
-  else
-    url += "&";
+  std::string paramsStr = std::string(params.substr(params.front() == '&' || params.front() == '?' ? 1 : 0));
+  while (paramsStr.find('=') != std::string::npos)
+  {
+    size_t keyValueDeviderPos = paramsStr.find('=');
+    size_t ampersandPos = paramsStr.find('&');
+    std::string paramKey = paramsStr.substr(0, keyValueDeviderPos);
+
+    // check if param key is already in url
+    if (url.find('?' + paramKey + '=') == std::string::npos && url.find('&' + paramKey + '=') == std::string::npos)
+    {
+      std::string paramValue;
+      if (ampersandPos == std::string::npos)
+        paramValue = paramsStr.substr(keyValueDeviderPos + 1);
+      else
+        paramValue = paramsStr.substr(keyValueDeviderPos + 1, ampersandPos);
+
+      url += url.find_first_of('?') == std::string::npos ? '?' : '&';
+      url += paramKey + '=' + paramValue;
+    }
 
-  url += params.substr(params.front() == '&' || params.front() == '?' ? 1 : 0);
+    if (ampersandPos != std::string::npos)
+      paramsStr = paramsStr.substr(ampersandPos + 1);
+    else
+      paramsStr.clear();
+  }
```

Comment:
i think new solution its better,
i tried optimize a bit the code, but untested please apply and test these changes:

```c++
  params.remove_prefix(params.front() == '&' || params.front() == '?' ? 1 : 0);
  size_t keyValDividerPos;
  while ((keyValDividerPos = params.find('=')) != std::string::npos)
  {
    size_t ampersandPos = params.find('&');
    std::string paramKey{params.substr(0, keyValDividerPos)};

    // if param key is not in url add it
    if (url.find('?' + paramKey + '=') == std::string::npos &&
        url.find('&' + paramKey + '=') == std::string::npos)
    {
      url += url.find_first_of('?') == std::string::npos ? '?' : '&';
      url += paramKey + '=';
      url += params.substr(keyValDividerPos + 1, ampersandPos);
    }

    if (ampersandPos != std::string::npos)
      params = params.substr(ampersandPos + 1);
    else
      break;
  }
```